boa/interpret.py

```diff
@@ -249,8 +254,10 @@ def loads_partial(
     if dedent:
         source_code = textwrap.dedent(source_code)
 
-    version = _detect_version(source_code)
-    if version is not None and version != vyper.__version__:
+    specifier_set = detect_version_specifier_set(source_code)
+    # Use VVM only if the installed version is not in the specifier set
+    if specifier_set is not None and not specifier_set.contains(vyper.__version__):
+        version = detect_vyper_version_from_source(source_code)
```

Comment:
this results in two searches of the source code. i think we can use vvm's _pick_vyper_version instead?